modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/api/SessionImplTest.java

```diff
@@ -0,0 +1,220 @@
+import static org.apache.ignite.lang.ErrorGroups.Sql.SESSION_CLOSED_ERR;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.ignite.internal.sql.engine.AsyncSqlCursor;
+import org.apache.ignite.internal.sql.engine.QueryProcessor;
+import org.apache.ignite.internal.sql.engine.SqlQueryType;
+import org.apache.ignite.internal.sql.engine.session.SessionId;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.util.AsyncCursor.BatchedResult;
+import org.apache.ignite.internal.util.IgniteSpinBusyLock;
+import org.apache.ignite.sql.BatchedArguments;
+import org.apache.ignite.sql.Session.SessionBuilder;
+import org.apache.ignite.sql.async.AsyncResultSet;
+import org.apache.ignite.tx.IgniteTransactions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * Tests to verify {@link SessionImpl}.
+ */
+@SuppressWarnings({"resource", "ThrowableNotThrown"})
+@ExtendWith(MockitoExtension.class)
+class SessionImplTest extends BaseIgniteAbstractTest {
+    private final ConcurrentMap<SessionId, SessionImpl> sessions = new ConcurrentHashMap<>();
+
+    private final AtomicLong clock = new AtomicLong();
+
+    @Mock
+    private QueryProcessor queryProcessor;
+
+    @BeforeEach
+    void setUp() {
+        clock.set(1L);
+
+        when(queryProcessor.closeSession(any()))
+                .thenReturn(CompletableFuture.completedFuture(null));
+    }
+
+    @AfterEach
+    void cleanUp() {
+        sessions.values().forEach(SessionImpl::close);
+
+        sessions.clear();
+    }
+
+    @Test
+    void sessionExpiredAfterIdleTimeout() {
+        SessionImpl session = newSession(2);
+        assertThat(session.expired(), is(false));
+
+        //period is small to expire session
+        clock.addAndGet(1);
+        assertThat(session.expired(), is(false));
+
+        //period is enough to session expired
+        clock.addAndGet(2);
+        assertThat(session.expired(), is(true));
+    }
+
+    @RepeatedTest(10)
```

Comment:
What reason to run it few times? 
If it really required then let's add some comment about the reason

Reply:
no reason, replaced with regular test